src/Compilers/CSharp/Portable/Syntax/Syntax.xml

```diff
@@ -2231,8 +2241,10 @@
       <Kind Name="OpenParenToken"/>
     </Field>
     <!-- Declaration and Initializers are mutually exclusive. -->
-    <Field Name="Declaration" Type="VariableDeclarationSyntax" Optional="true"/>
-    <Field Name="Initializers" Type="SeparatedSyntaxList&lt;ExpressionSyntax&gt;"/>
+    <Choice>
+      <Field Name="Declaration" Type="VariableDeclarationSyntax" Optional="true"/>
```

Comment:
>Optional="true" [](start = 65, length = 15)

Is this still needed?

Reply:
in this *yes*.  because for a for loop, it isn't: 

` | 'for' '(' ( variable_declaration | initializers ) ...`  instead, it is:
` | 'for' '(' ( variable_declaration? | initializers ) ...`

(although, i suppose this could be the former case, with initializers also resolving to empty).  but i wanted to indicate that this was not just "either or" but also that "neither" is acceptable.  That's different from, say, lambdas.  In that case you need an expr *or* a block.  You can't have "neither". But with for-statement, "neither" is acceptable :)  

Does that make sense?

Reply:
other option would be to make the 'Choice' itself optional.  any personal preferences.  they're all about equal in terms of expressivity...

Reply:
I would probably have gone for making `Choice` optional, or adding a `<None>` entry. But current approach seems okay (it just seems a bit arbitrary that the `optional` is on the variable_declaration).